sdk/src/main/java/com/dailymotion/android/player/sdk/PlayerWebView.java

```diff
@@ -528,24 +534,54 @@ public void setFullscreenButton(boolean fullScreen) {
 
     public PlayerWebView(Context context) {
         super(context);
+        new AdIdTask(context, this).execute();
     }
 
     public PlayerWebView(Context context, AttributeSet attrs) {
         super(context, attrs);
+        new AdIdTask(context, this).execute();
     }
 
     public PlayerWebView(Context context, AttributeSet attrs, int defStyleAttr) {
         super(context, attrs, defStyleAttr);
+        new AdIdTask(context, this).execute();
+    }
+
+    private void continueInitialize() {
+        handler.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                if (mHasAdId) {
+                    initialize(mBaseUrl, mQueryParameters, mHttpHeaders);
+                } else {
+                    continueInitialize();
+                }
+            }
+        },500);
+    }
+
+    @Override
+    public void onResult(String result) {
+        mAdId = result;
+        mHasAdId = true;
     }
 
     public void initialize(String baseUrl, Map<String, String> queryParameters, Map<String, String> httpHeaders) {
 
+        if (!mHasAdId) {
+            mBaseUrl = baseUrl;
+            mQueryParameters = queryParameters;
+            mHttpHeaders = httpHeaders;
+            continueInitialize();
+            return;
+        }
+
```

Comment:
what about starting the AsyncTask here ? And capturing the parameters in a closure instead of adding them as fields ?